src/renderer/components/ScreenSharePicker.tsx

```diff
@@ -380,35 +444,83 @@ function ModalComponent({
-                        const conn = [...MediaEngineStore.getMediaEngine().connections][1];
-                        if (conn && conn.videoStreamParameters.length > 0) {
-                            const height = Number(settings.resolution);
-                            const width = Math.round(height * (16 / 9));
-                            Object.assign(conn.videoStreamParameters[0], {
-                                maxFrameRate: Number(settings.fps),
-                                maxPixelCount: width * height,
-                                maxBitrate: 8000000,
-                                maxResolution: {
-                                    type: "fixed",
-                                    width,
-                                    height
-                                }
-                            });
+                        try {
+                            var conn = [...MediaEngineStore.getMediaEngine().connections].find(
+                                connection => connection.streamUserId === UserStore.getCurrentUser().id
+                            );
+                        } catch {
+                            console.log("No current stream.");
+                        }
+                        try {
+                            if (conn) {
+                                const track = conn.input.stream.getVideoTracks()[0];
+                                const frameRate = Number(settings.fps);
+                                const height = Number(settings.resolution);
+                                const width = Math.round(height * (16 / 9));
+                                var constraints = track.getConstraints();
+                                const newConstraints = {
+                                    ...constraints,
+                                    frameRate,
+                                    width: { min: 640, ideal: width, max: 2560 },
+                                    height: { min: 480, ideal: height, max: 1440 },
+                                    advanced: [{ width: width, height: height }],
+                                    resizeMode: "none"
+                                };
+                                track.applyConstraints(newConstraints).then(() => {
+                                    console.log("Applied constraints from ScreenSharePicker successfully.");
+                                    console.log("New constraints:", track.getConstraints());
+                                });
+
+                                // changing stream quality description
+                                conn.videoStreamParameters[0].maxFrameRate = Number(settings.fps);
+                                conn.videoStreamParameters[0].maxResolution.height = Number(settings.resolution);
+                                conn.videoStreamParameters[0].maxResolution.width = Math.round(height * (16 / 9));
+                            }
+                        } catch {
```

Comment:
this try catch seems duplicated, and the error message misleading

merge them and add more logical error handling, or remove them altogether